src/bootstrap/src/core/config/config.rs

```diff
@@ -1767,6 +1767,29 @@ impl Config {
                 std_features: std_features_toml,
             } = rust;
 
+            // Before something like [Enable debug assertions on alt
+            // builds](https://github.com/rust-lang/rust/pull/131077) lands, alt rustc builds do
+            // *not* have rustc debug assertions enabled. We must not download an CI alt rustc if we
+            // need rustc to have debug assertions (e.g. for crashes test suite).
+            //
+            // Note that `rust.debug = true` (currently) implies `rust.debug-assertions = true`!
+            //
+            // This relies also on the fact that global default for `download-rustc` will be `false`
+            // if it's not explicitly set.
+            if matches!(rustc_debug_assertions, Some(true))
+                || (matches!(debug, Some(true)) && rustc_debug_assertions.is_none())
+            {
```

Comment:
I don't like this but this is the simplest thing I could think of without fixing our config layering